src/coreclr/src/vm/genericdict.cpp

```diff
@@ -1245,6 +1493,13 @@ Dictionary::PopulateEntry(
 
         if ((slotIndex != 0) && !IsCompilationProcess())
         {
+            // Lock is needed because dictionary pointers can get updated during dictionary size expansion
+            CrstHolder ch(&SystemDomain::SystemModule()->m_DictionaryCrst);
```

Comment:
Do you need to take this lock even when we are not resizing the dictionary (typical case)?

Reply:
No. I missed that. We can move the lock to the point where we actually resize